airbyte-integrations/connector-templates/source-python/source_{{snakeCase name}}/source.py.hbs

```diff
@@ -23,21 +23,98 @@ SOFTWARE.
 
 
 class Source{{properCase name}}(Source):
-
     def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
-        raise Exception("unimplemented")
+        """
+        Tests if the input configuration can be used to successfully connect to the integration
+            e.g: if a provided Stripe API token can be used to connect to the Stripe API.
+
+        :param logger: Logging object to display debug/info/error to the logs
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            # Not Implemented
+
+            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+        except Exception as e:
+            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
 
     def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
-        # discover the schema with the provided config
-        raise Exception("unimplemented")
+        """
+        Returns an AirbyteCatalog representing the available streams and fields in this integration.
+        For example, given valid credentials to a Postgres database,
+        returns an Airbyte catalog where each postgres table is a stream, and each table column is a field.
+
+        :param logger: Logging object to display debug/info/error to the logs
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+        the properties of the spec.json file
+
+        :return: AirbyteCatalog is an object describing a list of all available streams in this source.
+            A stream is an AirbyteStream object that includes:
+            - its stream name (or table name in the case of Postgres)
+            - json_schema providing the specifications of expected schema for this stream (a list of columns described
+            by their names and types)
+        """
+        streams = []
+
+        stream_name = "TableName"  # Example
+        json_schema = {  # Example
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "type": "object",
+            "properties": {"columnName": {"type": "string"}},
+        }
+
+        # Not Implemented
+
+        streams.append(AirbyteStream(name=stream_name, json_schema=json_schema))
+        return AirbyteCatalog(streams=streams)
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
+        """
+        Returns a generator of the AirbyteMessages generated by reading the source with the given configuration,
+        catalog, and state.
+
+        :param logger: Logging object to display debug/info/error to the logs
```

Comment:
worth mentioning with the logger that if they _don't_ use the logger that their logs will not be accessible via airbyte.